### PR TITLE
Adds bdbag recipe

### PR DESCRIPTION
**Pull request by sterlingbaldwin**
Checklist

- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vend other packages
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by sterlingbaldwin** [comment]
@mikedarcy I just set this up to get the process started. Please acknowledge you want to be a package maintainer. If you would like I can take my name off and have you the sole maintainer since bdbag is your project, I just wanted to help setting this up.

**Comment by mikedarcy** [comment]
@sterlingbaldwin Thanks for setting it up. I had started in on it but got pulled into something and wasn't able to finish.  I am fine with being a package maintainer and also having you be listed as a maintainer as well. 

I think you need to add all possible dependencies since some are runtime only and not explicitly declared. I dont know what their equivalent conda forge names are.

```
"boto3>=1.9.5",
"botocore", 
"awscli",
"globus_sdk>=1.6.0"
```

**Comment by sterlingbaldwin** [comment]
Thanks @mikedarcy, I had some extra time and thought I would take care of this (also I want my Hacktoberfest PRs!).

File: recipes/bdbag/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  imports:
+    - bdbag
+  commands:
+    bdbag -h
+
+about:
+  home: https://github.com/fair-research/bdbag
+  license: Apache-2.0
+  license_family: Apache
+  license_file: LICENSE.txt
```
Comment (saraedum) [inline recipes/bdbag/meta.yaml]: The tarball seems to contain the `LICENSE`. I don't think you need to ship this explicitly with the recipe.

File: recipes/bdbag/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+{% set name = "bdbag" %}
+{% set version = "1.5.6" %}
+{% set hash_type = "md5" %}
```
Comment (saraedum) [inline recipes/bdbag/meta.yaml]: We usually use sha256 hashes. Any reason why you must use md5 here?

File: recipes/bdbag/meta.yaml (hunk 1 above)
Comment (sterlingbaldwin) [reply recipes/bdbag/meta.yaml]: Apparently it needs to be there or the linting check fails.

**Comment by sterlingbaldwin** [comment]
The full test suite requires additional data from other optional github repos, packaging along the test data would be a bit more difficult and probably overkill. @mikedarcy would know better then me.

File: recipes/bdbag/meta.yaml (hunk 1 above)
Comment (saraedum) [reply recipes/bdbag/meta.yaml]: Right. Needs to be in that section but you don't need to package the file. I believe the file is called `LICENSE` in the tarball. So, just calling it `LICENSE` should pick up the file from the tarball.

**Comment by saraedum** [comment]
> The full test suite requires additional data from other optional github repos, packaging along the test data would be a bit more difficult and probably overkill. @mikedarcy would know better then me.

Sure. No problem. I was just curious what doesn't work.

File: recipes/bdbag/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
+{% set name = "bdbag" %}
+{% set version = "1.5.6" %}
+{% set hash_type = "sha256" %}
```
Comment (saraedum) [inline recipes/bdbag/meta.yaml]: Not really important, but I'd probably just inline this. It's a bit nonstandard to push this into a variable I believe.

```suggestion
```

File: recipes/bdbag/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+{% set hash_value = "ae057fec9640442ad039fe50b60afa1d71c0109f9163d182f4f4b27774bbc8b3" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  {{ hash_type }}: {{ hash_value }}
```
Comment (saraedum) [inline recipes/bdbag/meta.yaml]:
```suggestion
  sha256: {{ hash_value }}
```

File: recipes/bdbag/meta.yaml (hunk 3 above)
Comment (dopplershift) [reply recipes/bdbag/meta.yaml]: Same thing with `hash_value`

Comment (saraedum) [reply recipes/bdbag/meta.yaml]: True. I suppose that the automatic updates (regro-cf-autotick-bot) won't update the SHA256 automatically for you if you are using any variables there.

File: recipes/bdbag/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ hash_value }}
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
```
Comment (bernt-matthias) [inline recipes/bdbag/meta.yaml:17]:
```suggestion
  script: "{{ PYTHON }} -m pip install . --no-deps -vv
```

File: recipes/bdbag/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,63 @@
+{% set name = "bdbag" %}
+{% set version = "1.5.6" %}
+{% set hash_type = "sha256" %}
+{% set hash_value = "ae057fec9640442ad039fe50b60afa1d71c0109f9163d182f4f4b27774bbc8b3" %}
```
Comment (bernt-matthias) [inline recipes/bdbag/meta.yaml:4]:
```suggestion
{% set sha256 = "ae057fec9640442ad039fe50b60afa1d71c0109f9163d182f4f4b27774bbc8b3" %}
```

File: recipes/bdbag/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ hash_value }}
```
Comment (bernt-matthias) [inline recipes/bdbag/meta.yaml:12]:
```suggestion
  sha256: {{ sha256 }}
```

File: recipes/bdbag/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    - bdbag
+  commands:
+    bdbag -h
+
+about:
+  home: https://github.com/fair-research/bdbag
+  summary: 'The bdbag utilities are a collection of software programs for working with BagIt packages that conform to the BDBag and Bagit/RO profiles.'
+  license: Apache-2.0
+  license_family: Apache
+  license_file: LICENSE.txt
```
Comment (bernt-matthias) [inline recipes/bdbag/meta.yaml:50]:
```suggestion
  license_file: LICENSE
```

**Review by bernt-matthias — COMMENTED** [review]
Hi @sterlingbaldwin and @mikedarcy . How about reviving this PR? I added some suggestion mirroring the review comments. 

Would be great to get this in conda-forge.

**Comment by mikedarcy** [comment]
I've no problem with this. However, @sterlingbaldwin did most of the work in the repo referenced by this PR, so I think it would be best to get that repo updated with these patches and reopen the PR. Otherwise, I suppose we should start over with a new repo and pull request.

**Comment by bernt-matthias** [comment]
OK. Can you reopen it? Otherwise I could open a new one with the commits from here.

**Comment by mikedarcy** [comment]
Since we haven't heard from @sterlingbaldwin and this PR references his repo fork, I think it might just be better to open a new one. Let's give him a final chance to weigh in. If we don't hear from him in the next day or so, Matthias please feel free to proceed with a new PR.  Also, thanks for picking this back up and driving. I appreciate the help.

**Comment by bernt-matthias** [comment]
Ping.. https://github.com/conda-forge/staged-recipes/pull/13801 .. just need to add my suggestions.

**Comment by sterlingbaldwin** [comment]
hey sorry y'all I wasnt checking my notifications, but it looks like this got sorted out.

**Pull request closed without merging**